### PR TITLE
docs: add documentation for .tailcallrc.graphql

### LINKED ISSUE #740: Add documentation to each operator and it fields
`.tailcallrc.graphql` doesn't have any documentation. We should add some documentation using `"""` for each operator and field to make sure that there is some helpful text available with auto-complete.

### PR DESCRIPTION
**Pull request by neo773**
Adds documentation for directives properties in .taillrc.graphql

https://github.com/tailcallhq/tailcall/assets/62795688/441b9109-e019-40dc-8a42-f52f69330585

/fixes #740
/claim #740

### CODE REVIEW

**Review by ologbonowiwi — CHANGES_REQUESTED** [review]
Nice work documenting the properties! It looks really good.

Although, it's lacking docs for the operators itself (e.g.: `@server`, `@upstream`), and also to the operations and fields below line [194](https://github.com/tailcallhq/tailcall/pull/741/files#diff-df81d69338b9c82f79a6c2b67791c6865740f1c1b8c6bac6af6b8021e2399dc3R194).

**Comment by neo773** [comment]
@ologbonowiwi  @tusharmath 
Hmm do we really need it? seemed self explanatory

**Comment by ologbonowiwi** [comment]
IMO, we need it. I usually see newcomers have confusion about `@server` and `@upstream`.

As for the other fields, they are self-explanatory to us who are familiar with the project, but I'm not so sure about people who are having their first experience with tailcall

**Comment by neo773** [comment]
> I agree @ologbonowiwi - we should add documentation for everything. Secondly and more importantly, the documentation should be a little more detailed. Currently it looks like GPT generated low effort work (sorry 🙈 )

The original version looked way too verbose for the tiny detail box, that's why I went with a concise version
Will rewrite it

**Comment by neo773** [comment]
@tusharmath 
Pushed a new commit, rewrote it all

File: examples/.tailcallrc.graphql [hunk 1]
```diff
@@ -22,35 +123,312 @@ enum HttpVersion {
...
+}
+
+type User {
+  name: String
+  age: Int
+}
+```
+"""
 directive @const(data: JSON) on FIELD_DEFINITION
 directive @graphQL(
```
Comment (ologbonowiwi) [inline examples/.tailcallrc.graphql:433]: I guess we need docs also for these (and it's respective fields)

**Review by ologbonowiwi — CHANGES_REQUESTED** [review]

File: examples/.tailcallrc.graphql [hunk 2]
```diff
@@ -1,19 +1,120 @@
...
+  """
+  `responseHeaders` appends headers to all server responses, aiding cross-origin requests or extra headers for downstream services.
+
+  The responseHeader is a key-value pair array. These headers are included in every server response. Useful for headers like Access-Control-Allow-Origin for cross-origin requests, or additional headers like X-Allowed-Roles for downstream services.
+
+  ```graphql
+  schema @server(responseHeaders: [{key: "X-Allowed-Roles", value: "admin,user"}]) {
+    query: Query
+    mutation: Mutation
+  }
```
Comment (tusharmath) [inline examples/.tailcallrc.graphql:81]: Can drop the code from the documentation.

File: examples/.tailcallrc.graphql (hunk 1 above)
Comment (tusharmath) [reply examples/.tailcallrc.graphql:224]: Agree.

**Review by tusharmath — APPROVED** [review]

**Pull request merged**